<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>systemPipeR: Data Visualizations • SPtools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="systemPipeR: Data Visualizations">
<meta property="og:description" content="SPtools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/systemPipeR_dataViz.html">systemPipeR: Data Visualizations</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="systemPipeR_dataViz_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>systemPipeR: Data Visualizations</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 21 April, 2021</h4>
      
      
      <div class="hidden name"><code>systemPipeR_dataViz.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!--
- Compile from command-line
Rscript -e "rmarkdown::render('systemPipeR_dataViz.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeR_dataViz.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeR.knit.md 2; Rscript -e "rmarkdown::render('systemPipeR_dataViz.Rmd', c('BiocStyle::pdf_document'))"
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="data-visualization-with-systempiper" class="section level1">
<h1 class="hasAnchor">
<a href="#data-visualization-with-systempiper" class="anchor"></a>Data Visualization with <code>systemPipeR</code>
</h1>
<div id="metadata-and-reads-counting-information" class="section level2">
<h2 class="hasAnchor">
<a href="#metadata-and-reads-counting-information" class="anchor"></a>Metadata and Reads Counting Information</h2>
<p>The first step is importing the <code>targets</code> file and raw reads counting table. - The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow. - The raw reads counting table represents all the reads that map to gene (row) for each sample (columns).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Targets file</span>
<span class="va">targetspath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">targetspath</span>, comment <span class="op">=</span> <span class="st">"#"</span><span class="op">)</span>
<span class="va">cmp</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">targetspath</span>, format <span class="op">=</span> <span class="st">"matrix"</span>, delim <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span>
<span class="co">## Count table file</span>
<span class="va">countMatrixPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"countDFeByg.xls"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">countMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">countMatrixPath</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/showDT.html">showDT</a></span><span class="op">(</span><span class="va">countMatrix</span><span class="op">)</span></code></pre></div>
</div>
<div id="data-transformation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-transformation" class="anchor"></a>Data Transformation</h2>
<p>For gene differential expression, raw counts are required, however for data visualization or clustering, it can be useful to work with transformed count data. <code>exploreDDS</code> function is convenience wrapper to transform raw read counts using the <a href="@Love2014-sh"><code>DESeq2</code></a> package transformations methods. The input file has to contain all the genes, not just differentially expressed ones.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exploredds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exploreDDS.html">exploreDDS</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, 
    transformationMethod <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span>
<span class="va">exploredds</span></code></pre></div>
<p>Users are strongly encouraged to consult the <a href="@Love2014-sh"><code>DESeq2</code></a> vignette for more detailed information on this topic and how to properly run <code>DESeq2</code> on data sets with more complex experimental designs.</p>
</div>
<div id="scatterplot" class="section level2">
<h2 class="hasAnchor">
<a href="#scatterplot" class="anchor"></a>Scatterplot</h2>
<p>To decide which transformation to choose, we can visualize the transformation effect comparing two samples or a grid of all samples, as follows:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">exploreDDSplot</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"M12A"</span>, 
    <span class="st">"A12A"</span><span class="op">)</span>, scattermatrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">exploreDDSplot</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"M12A"</span>, 
    <span class="st">"M12A"</span>, <span class="st">"A12A"</span>, <span class="st">"A12A"</span><span class="op">)</span>, scattermatrix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>The scatterplots are created using the log2 transform normalized reads count, variance stabilizing transformation (VST) (<span class="citation">Anders and Huber (2010)</span>), and regularized-logarithm transformation or <code>rlog</code> (<span class="citation">Love, Huber, and Anders (2014)</span>).</p>
</div>
<div id="hierarchical-clustering-dendrogram" class="section level2">
<h2 class="hasAnchor">
<a href="#hierarchical-clustering-dendrogram" class="anchor"></a>Hierarchical Clustering Dendrogram</h2>
<p>The following computes the sample-wise correlation coefficients using the <code><a href="https://rdrr.io/r/stats/cor.html">stats::cor()</a></code> function from the transformed expression values. After transformation to a distance matrix, hierarchical clustering is performed with the <code><a href="https://rdrr.io/r/stats/hclust.html">stats::hclust</a></code> function and the result is plotted as a dendrogram, as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">hclustplot</a></span><span class="op">(</span><span class="va">exploredds</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">hclustplot</a></span><span class="op">(</span><span class="va">exploredds</span>, method <span class="op">=</span> <span class="st">"spearman"</span>, savePlot <span class="op">=</span> <span class="cn">TRUE</span>, filePlot <span class="op">=</span> <span class="st">"cor.pdf"</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="hierarchical-clustering-heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#hierarchical-clustering-heatmap" class="anchor"></a>Hierarchical Clustering HeatMap</h2>
<p>This function performs hierarchical clustering on the transformed expression matrix generated within the <code>DESeq2</code> package. It uses, by default, a <code>Pearson</code> correlation-based distance measure and complete linkage for cluster join. If <code>samples</code> selected in the <code>clust</code> argument, it will be applied the <code><a href="https://rdrr.io/r/stats/dist.html">stats::dist()</a></code> function to the transformed count matrix to get sample-to-sample distances. Also, it is possible to generate the <code>pheatmap</code> or <code>plotly</code> plot format.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Samples plot</span>
<span class="fu"><a href="../reference/exploreDDS.html">heatMaplot</a></span><span class="op">(</span><span class="va">exploredds</span>, clust <span class="op">=</span> <span class="st">"samples"</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">heatMaplot</a></span><span class="op">(</span><span class="va">exploredds</span>, clust <span class="op">=</span> <span class="st">"samples"</span>, plotly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>If <code>ind</code> selected in the <code>clust</code> argument, it is necessary to provide the list of differentially expressed genes for the <code>exploredds</code> subset.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Individuals genes identified in DEG analysis DEG analysis with `systemPipeR`</span>
<span class="va">degseqDF</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/run_DESeq2.html">run_DESeq2</a></span><span class="op">(</span>countDF <span class="op">=</span> <span class="va">countMatrix</span>, targets <span class="op">=</span> <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
    independent <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">DEG_list</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/filterDEGs.html">filterDEGs</a></span><span class="op">(</span>degDF <span class="op">=</span> <span class="va">degseqDF</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Fold <span class="op">=</span> <span class="fl">2</span>, FDR <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span>
<span class="co">### Plot</span>
<span class="fu"><a href="../reference/exploreDDS.html">heatMaplot</a></span><span class="op">(</span><span class="va">exploredds</span>, clust <span class="op">=</span> <span class="st">"ind"</span>, DEGlist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">DEG_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="principal-component-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal Component Analysis</h2>
<p>This function plots a Principal Component Analysis (PCA) from transformed expression matrix. This plot shows samples variation based on the expression values and identifies batch effects.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">PCAplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">PCAplot</a></span><span class="op">(</span><span class="va">exploredds</span>, save <span class="op">=</span> <span class="cn">TRUE</span>, filePlot <span class="op">=</span> <span class="st">"pca.pdf"</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="multidimensional-scaling-with-mdsplot" class="section level2">
<h2 class="hasAnchor">
<a href="#multidimensional-scaling-with-mdsplot" class="anchor"></a>Multidimensional scaling with <code>MDSplot</code>
</h2>
<p>This function computes and plots multidimensional scaling analysis for dimension reduction of count expression matrix. Internally, it is applied the <code><a href="https://rdrr.io/r/stats/dist.html">stats::dist()</a></code> function to the transformed count matrix to get sample-to-sample distances.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">MDSplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">TRUE</span>, save <span class="op">=</span> <span class="cn">TRUE</span>, filePlot <span class="op">=</span> <span class="st">"MDSplot.pdf"</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="dimension-reduction-with-glmplot" class="section level2">
<h2 class="hasAnchor">
<a href="#dimension-reduction-with-glmplot" class="anchor"></a>Dimension Reduction with <code>GLMplot</code>
</h2>
<p>This function computes and plots generalized principal components analysis for dimension reduction of count expression matrix.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exploredds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exploreDDS.html">exploreDDS</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, 
    transformationMethod <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">GLMplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="../reference/exploreDDS.html">GLMplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span>, savePlot <span class="op">=</span> <span class="cn">TRUE</span>, filePlot <span class="op">=</span> <span class="st">"GLM.pdf"</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="ma-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#ma-plot" class="anchor"></a>MA plot</h2>
<p>This function plots log2 fold changes (y-axis) versus the mean of normalized counts (on the x-axis). Statistically significant features are colored.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">MAplot</a></span><span class="op">(</span><span class="va">exploredds</span><span class="op">)</span></code></pre></div>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="t-distributed-stochastic-neighbor-embedding-with-tsneplot" class="section level2">
<h2 class="hasAnchor">
<a href="#t-distributed-stochastic-neighbor-embedding-with-tsneplot" class="anchor"></a>t-Distributed Stochastic Neighbor embedding with <code>tSNEplot</code>
</h2>
<p>This function computes and plots t-Distributed Stochastic Neighbor embedding (t-SNE) analysis for unsupervised nonlinear dimensionality reduction of count expression matrix. Internally, it is applied the <code><a href="https://rdrr.io/pkg/Rtsne/man/Rtsne.html">Rtsne::Rtsne()</a></code> function, using the exact t-SNE computing with <code>theta=0.0</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">tSNEplot</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, perplexity <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div id="volcano-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#volcano-plot" class="anchor"></a>Volcano plot</h2>
<p>A simple function that shows statistical significance (<code>p-value</code>) versus magnitude of change (<code>log2 fold change</code>).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDS.html">volcanoplot</a></span><span class="op">(</span><span class="va">degseqDF</span>, comparison <span class="op">=</span> <span class="st">"M12-A12"</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Fold <span class="op">=</span> <span class="fl">1</span>, FDR <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, genes <span class="op">=</span> <span class="st">"ATCG00280"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version information</h1>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2021-02-04 r79940)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /home/dcassol/src/R-devel/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] runPlots_0.9     BiocStyle_2.19.2
## 
## loaded via a namespace (and not attached):
##   [1] Rtsne_0.15                  colorspace_2.0-0           
##   [3] ggtree_2.5.2                ellipsis_0.3.1             
##   [5] rprojroot_2.0.2             XVector_0.31.1             
##   [7] GenomicRanges_1.43.4        fs_1.5.0                   
##   [9] aplot_0.0.6                 ggrepel_0.9.1              
##  [11] bit64_4.0.5                 AnnotationDbi_1.53.1       
##  [13] fansi_0.4.2                 codetools_0.2-18           
##  [15] splines_4.1.0               cachem_1.0.4               
##  [17] geneplotter_1.69.0          knitr_1.32                 
##  [19] jsonlite_1.7.2              annotate_1.69.2            
##  [21] png_0.1-7                   pheatmap_1.0.12            
##  [23] BiocManager_1.30.12         compiler_4.1.0             
##  [25] httr_1.4.2                  rvcheck_0.1.8              
##  [27] assertthat_0.2.1            Matrix_1.3-2               
##  [29] fastmap_1.1.0               lazyeval_0.2.2             
##  [31] formatR_1.9                 htmltools_0.5.1.1          
##  [33] tools_4.1.0                 gtable_0.3.0               
##  [35] glue_1.4.2                  GenomeInfoDbData_1.2.4     
##  [37] dplyr_1.0.5                 Rcpp_1.0.6                 
##  [39] Biobase_2.51.0              jquerylib_0.1.3            
##  [41] pkgdown_1.6.1               vctrs_0.3.7                
##  [43] Biostrings_2.59.2           ape_5.4-1                  
##  [45] nlme_3.1-152                xfun_0.22                  
##  [47] stringr_1.4.0               lifecycle_1.0.0            
##  [49] XML_3.99-0.6                zlibbioc_1.37.0            
##  [51] MASS_7.3-53.1               scales_1.1.1               
##  [53] ragg_1.1.2                  MatrixGenerics_1.3.1       
##  [55] parallel_4.1.0              SummarizedExperiment_1.21.3
##  [57] RColorBrewer_1.1-2          yaml_2.2.1                 
##  [59] memoise_2.0.0               ggplot2_3.3.3              
##  [61] sass_0.3.1                  reshape_0.8.8              
##  [63] stringi_1.5.3               RSQLite_2.2.6              
##  [65] genefilter_1.73.1           S4Vectors_0.29.15          
##  [67] desc_1.3.0                  tidytree_0.3.3             
##  [69] BiocGenerics_0.37.1         BiocParallel_1.25.5        
##  [71] GenomeInfoDb_1.27.11        rlang_0.4.10               
##  [73] pkgconfig_2.0.3             systemfonts_1.0.1          
##  [75] matrixStats_0.58.0          bitops_1.0-6               
##  [77] evaluate_0.14               lattice_0.20-41            
##  [79] purrr_0.3.4                 htmlwidgets_1.5.3          
##  [81] treeio_1.15.6               patchwork_1.1.1            
##  [83] bit_4.0.4                   tidyselect_1.1.0           
##  [85] GGally_2.1.1                plyr_1.8.6                 
##  [87] magrittr_2.0.1              bookdown_0.21              
##  [89] DESeq2_1.31.18              R6_2.5.0                   
##  [91] IRanges_2.25.8              generics_0.1.0             
##  [93] DelayedArray_0.17.10        DBI_1.1.1                  
##  [95] pillar_1.6.0                survival_3.2-10            
##  [97] KEGGREST_1.31.1             RCurl_1.98-1.3             
##  [99] tibble_3.1.0                crayon_1.4.1               
## [101] utf8_1.2.1                  plotly_4.9.3               
## [103] rmarkdown_2.7               locfit_1.5-9.4             
## [105] grid_4.1.0                  data.table_1.14.0          
## [107] blob_1.2.1                  digest_0.6.27              
## [109] glmpca_0.2.0                xtable_1.8-4               
## [111] tidyr_1.1.3                 textshaping_0.3.3          
## [113] stats4_4.1.0                munsell_0.5.0              
## [115] viridisLite_0.4.0           bslib_0.2.4</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project is funded by NSF award <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1661152">ABI-1661152</a>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Anders2010-tp" class="csl-entry">
Anders, Simon, and Wolfgang Huber. 2010. <span>“Differential Expression Analysis for Sequence Count Data.”</span> <em>Genome Biol.</em> 11 (10): R106.
</div>
<div id="ref-Love2014-sh" class="csl-entry">
Love, Michael, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for <span class="nocase">RNA-seq</span> Data with <span>DESeq2</span>.”</span> <em>Genome Biol.</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Daniela Cassol, Ponmathi Ramasamy, Le Zhang, Thomas Girke.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
