<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>systemPipeTools: Data Visualizations • systemPipeTools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="systemPipeTools: Data Visualizations">
<meta property="og:description" content="systemPipeTools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">systemPipeTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/systemPipeTools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="systemPipeTools_files/header-attrs-2.7.12/header-attrs.js"></script><script src="systemPipeTools_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="systemPipeTools_files/jquery-3.5.1/jquery.min.js"></script><link href="systemPipeTools_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="systemPipeTools_files/datatables-binding-0.18/datatables.js"></script><link href="systemPipeTools_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="systemPipeTools_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="systemPipeTools_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script><link href="systemPipeTools_files/dt-ext-fixedcolumns-1.10.20/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="systemPipeTools_files/dt-ext-fixedcolumns-1.10.20/js/dataTables.fixedColumns.min.js"></script><link href="systemPipeTools_files/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet">
<script src="systemPipeTools_files/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script><link href="systemPipeTools_files/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet">
<script src="systemPipeTools_files/crosstalk-1.1.1/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>systemPipeTools: Data Visualizations</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 27 April, 2021</h4>
      
      
      <div class="hidden name"><code>systemPipeTools.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="data-visualization-with-systempiper" class="section level1">
<h1 class="hasAnchor">
<a href="#data-visualization-with-systempiper" class="anchor"></a>Data Visualization with <code>systemPipeR</code>
</h1>
<p><em>systemPipeTools</em> package extends the widely used <em><a href="https://systempipe.org/">systemPipeR</a></em> (SPR) <span class="citation">(H Backman and Girke 2016)</span> workflow environment with enhanced toolkit for data visualization, including utilities to automate the analysis of differentially expressed genes (DEGs). <em>systemPipeTools</em> provides functions for data transformation and data exploration via scatterplots, hierarchical clustering heatMaps, principal component analysis, multidimensional scaling, generalized principal components, t-Distributed Stochastic Neighbor embedding (t-SNE), and MA and volcano plots. All these utilities can be integrated with the modular design of the <em>systemPipeR</em> environment that allows users to easily substitute any of these features and/or custom with alternatives.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p><em><code>systemPipeTools</code></em> environment can be installed from the R console using the <a href="https://cran.r-project.org/web/packages/BiocManager/index.html"><em><code>BiocManager::install</code></em></a> command. The associated package <a href="http://bioconductor.org/packages/release/bioc/html/systemPipeR.html"><em><code>systemPipeR</code></em></a> can be installed the same way. The latter provides core functionalities for defining workflows, interacting with command-line software, and executing both R and/or command-line software, as well as generating publication-quality analysis reports.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"systemPipeTools"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"systemPipeR"</span><span class="op">)</span></code></pre></div>
<div id="loading-package-and-documentation" class="section level3">
<h3 class="hasAnchor">
<a href="#loading-package-and-documentation" class="anchor"></a>Loading package and documentation</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st">"systemPipeTools"</span><span class="op">)</span>  <span class="co"># Loads the package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span>help <span class="op">=</span> <span class="st">"systemPipeTools"</span><span class="op">)</span>  <span class="co"># Lists package info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/vignette.html">vignette</a></span><span class="op">(</span><span class="st">"systemPipeTools"</span><span class="op">)</span>  <span class="co"># Opens vignette</span></code></pre></div>
</div>
</div>
<div id="metadata-and-reads-counting-information" class="section level2">
<h2 class="hasAnchor">
<a href="#metadata-and-reads-counting-information" class="anchor"></a>Metadata and Reads Counting Information</h2>
<p>The first step is importing the <code>targets</code> file and raw reads counting table. - The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow. - The raw reads counting table represents all the reads that map to gene (row) for each sample (columns).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Targets file</span>
<span class="va">targetspath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"targets.txt"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">targetspath</span>, comment <span class="op">=</span> <span class="st">"#"</span><span class="op">)</span>
<span class="va">cmp</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">targetspath</span>, format <span class="op">=</span> <span class="st">"matrix"</span>, delim <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span>
<span class="co">## Count table file</span>
<span class="va">countMatrixPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"countDFeByg.xls"</span>, package <span class="op">=</span> <span class="st">"systemPipeR"</span><span class="op">)</span>
<span class="va">countMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="va">countMatrixPath</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="fu"><a href="../reference/showDT.html">showDT</a></span><span class="op">(</span><span class="va">countMatrix</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-567e95e7049f2e10b2f8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-567e95e7049f2e10b2f8">{"x":{"filter":"none","extensions":["FixedColumns","Scroller"],"data":[["AT1G01010","AT1G01020","AT1G01030","AT1G01040","AT1G01050","AT1G01060","AT1G01070","AT1G01073","AT1G01080","AT1G01090","AT1G01100","AT1G01110","AT1G01115","AT2G01008","AT2G01021","AT2G01023","AT3G01010","AT3G01015","AT3G01020","AT3G01030","AT3G01040","AT3G01050","AT3G01060","AT3G01070","AT3G01080","AT3G01085","AT3G01090","AT3G01100","AT3G01120","AT3G01130","AT3G01140","AT3G01150","AT3G01160","AT3G01170","AT4G00005","AT4G00020","AT4G00026","AT4G00030","AT4G00040","AT4G00050","AT4G00060","AT4G00070","AT4G00080","AT4G00090","AT4G00100","AT4G00110","AT4G00120","AT4G00124","AT4G00130","AT4G00140","AT5G01010","AT5G01015","AT5G01020","AT5G01030","AT5G01040","AT5G01050","AT5G01060","AT5G01070","AT5G01075","AT5G01080","AT5G01090","AT5G01100","AT5G01110","AT5G01120","AT5G01130","AT5G01140","AT5G01150","AT5G01160","ATCG00020","ATCG00040","ATCG00050","ATCG00070","ATCG00080","ATCG00120","ATCG00130","ATCG00140","ATCG00150","ATCG00160","ATCG00170","ATCG00180","ATCG00190","ATCG00210","ATCG00220","ATCG00270","ATCG00280","ATCG00300","ATCG00330","ATCG00340","ATCG00350","ATCG00360","ATCG00380","ATCG00420","ATCG00430","ATCG00440","ATCG00470","ATCG00480","ATCG00490","ATCG00500","ATCG00510","ATMG00010","ATMG00030","ATMG00040","ATMG00050","ATMG00060","ATMG00070","ATMG00080","ATMG00090","ATMG00110","ATMG00120","ATMG00130","ATMG00140","ATMG00150","ATMG00160","ATMG00170","ATMG00180","ATMG00200"],[57,23,41,180,60,26,0,0,98,331,230,6,0,28,6267,0,0,0,0,0,93,31,235,0,30,0,70,38,709,54,0,37,33,33,0,4,19,69,61,12,48,0,0,21,186,22,0,0,6,8,143,16,87,31,49,4,0,0,1,0,17,162,2,0,0,0,0,28,1917,9,0,5,15,52,34,36,17,7,7,5,37,6,3,105,283,4,52,147,192,1,2,7,21,12,54,180,856,8,10,2,60,11,0,0,8,7,5,6,3,1,0,0,5,3,18,0],[244,93,98,684,127,264,0,0,379,1027,432,10,0,68,3061,14,0,0,0,0,454,121,801,8,29,0,274,132,2358,140,14,126,238,131,0,25,89,106,176,138,359,0,0,59,384,78,0,0,9,6,442,26,227,170,125,7,0,4,28,0,103,1002,26,2,0,0,0,124,1696,28,3,23,50,210,127,46,53,8,39,49,145,5,83,300,723,34,103,412,480,6,21,97,312,44,339,1191,1029,13,1,0,40,56,0,3,18,18,72,27,0,4,7,1,31,15,59,3],[201,69,73,522,102,59,2,0,179,779,516,11,0,61,7559,6,0,0,0,0,364,108,553,6,151,0,315,87,2031,149,4,186,123,161,0,6,60,255,189,20,162,0,0,89,402,75,0,0,32,30,542,11,184,91,179,5,0,0,9,0,41,754,6,0,0,0,0,119,970,10,1,14,13,113,37,60,22,3,14,22,49,1,5,165,293,5,56,222,191,6,8,18,45,12,92,369,793,4,5,0,79,38,0,4,5,20,30,7,0,0,0,0,25,5,44,1],[169,126,58,664,166,56,1,0,456,1343,458,5,0,33,2203,10,0,1,0,0,578,167,674,2,14,0,373,126,2222,131,39,170,329,212,0,66,75,121,147,117,381,0,0,88,432,147,0,0,4,11,646,17,207,219,103,3,0,0,23,0,121,1185,27,2,0,0,0,127,661,19,1,59,53,190,167,42,88,14,94,68,208,16,75,212,641,45,135,530,520,10,24,147,347,34,423,1501,871,10,1,7,35,50,0,3,47,31,120,34,14,1,5,0,72,17,62,4],[365,107,94,585,125,8,6,0,287,647,468,4,0,48,3125,0,0,0,0,0,375,104,226,5,56,0,334,100,1674,242,2,138,133,110,0,16,41,127,158,23,275,0,0,56,403,81,0,0,28,30,606,4,165,97,150,7,0,0,2,0,62,819,11,2,0,0,0,103,2258,12,0,23,34,160,74,44,26,12,22,32,95,10,25,126,403,10,50,313,366,4,10,26,36,24,102,403,977,2,15,2,43,45,0,2,19,16,25,13,0,1,2,3,26,14,24,1],[229,88,156,680,303,26,6,0,538,1831,849,17,0,53,1168,3,0,1,0,0,735,146,366,6,97,0,349,203,2310,254,0,380,126,296,0,49,43,193,282,19,406,0,0,138,966,131,0,0,16,21,1031,8,463,106,427,10,2,0,5,1,89,2267,41,4,0,0,0,238,1232,6,0,24,56,277,66,33,54,11,55,61,186,3,29,250,375,23,58,509,455,1,10,63,112,15,347,984,1161,10,2,3,35,39,0,7,21,19,41,15,1,3,2,1,46,13,58,3],[41,18,9,162,116,5,2,0,102,350,522,3,0,39,12653,0,0,0,0,0,46,43,8,0,15,0,92,28,780,84,14,45,51,51,0,5,56,73,102,63,70,0,0,42,360,42,0,0,3,7,230,15,153,22,2,0,0,0,11,0,23,24,6,0,0,0,0,60,1677,4,0,5,4,37,8,41,2,0,1,11,18,2,4,63,186,2,19,121,116,0,0,7,8,2,19,73,592,0,2,1,89,19,0,2,4,8,11,5,0,1,0,0,8,6,27,2],[38,25,13,163,180,2,4,0,211,524,634,3,0,38,12386,1,0,0,0,0,76,55,59,0,25,0,139,50,1217,116,14,95,80,54,0,11,66,124,114,45,119,3,0,66,440,40,0,0,2,13,334,58,221,27,2,0,0,0,11,0,36,99,8,0,0,0,0,76,1184,6,0,1,7,42,20,18,16,1,6,15,22,2,2,59,190,4,13,152,168,0,3,11,9,1,30,163,681,2,1,0,35,37,0,4,9,4,9,3,0,1,2,0,16,8,26,2],[152,20,8,249,139,0,5,0,79,323,926,5,0,97,9799,0,0,0,0,0,80,52,13,0,72,0,148,74,1176,160,21,67,92,73,0,11,93,157,185,151,122,0,0,78,656,48,0,0,3,20,461,2,260,57,11,0,0,0,8,0,39,188,4,0,0,0,0,83,1201,0,0,1,11,41,7,23,5,0,6,2,15,2,0,56,132,2,23,75,122,0,2,4,10,4,22,93,475,4,0,0,117,31,0,5,1,5,12,7,2,1,3,0,19,9,33,0],[46,5,6,66,37,0,2,0,42,140,183,2,0,23,10386,1,0,0,0,0,23,21,1,0,7,0,48,19,270,27,7,14,20,22,0,1,12,21,48,8,43,0,0,22,113,12,0,0,1,3,109,10,50,29,2,3,0,0,1,0,13,42,0,0,0,0,0,33,1261,4,0,2,3,42,15,26,4,0,4,5,4,6,1,65,138,9,17,106,126,2,4,6,4,0,12,54,749,6,1,0,30,4,0,0,0,2,3,1,0,0,0,0,2,0,7,0],[294,88,36,697,236,0,5,0,150,600,1333,1,0,82,9414,3,0,2,0,0,162,112,14,0,57,0,236,126,1737,341,15,126,178,132,0,23,146,209,317,158,210,0,0,191,1005,46,0,0,12,22,731,0,423,100,7,0,0,0,12,0,68,277,5,0,0,0,0,128,1192,1,0,2,9,52,9,28,8,3,1,7,22,2,3,83,172,4,25,112,112,1,0,11,14,6,29,110,446,2,4,3,101,56,0,5,10,9,20,14,0,1,1,0,36,9,44,2],[405,151,147,1060,679,4,13,0,551,1693,2401,16,0,112,3265,4,0,0,0,0,260,292,41,2,33,0,468,280,2768,376,50,299,278,112,0,36,175,337,471,169,463,2,0,334,1831,107,0,0,12,13,1229,53,903,127,12,3,0,0,32,0,168,146,33,0,0,0,0,266,398,6,0,4,11,75,26,18,18,3,17,11,67,2,3,71,176,7,28,163,186,2,7,18,23,4,52,345,493,0,4,0,43,88,0,8,25,18,39,14,4,11,4,0,54,20,53,2],[117,43,32,338,362,89,9,0,250,800,762,7,0,142,6908,6,0,0,0,0,143,53,103,0,27,0,257,128,1998,110,52,130,65,95,0,19,38,245,362,472,298,0,0,74,703,51,0,0,5,23,449,6,732,62,2,0,0,0,12,0,46,42,2,2,0,0,0,157,1406,8,0,11,6,51,13,9,5,0,5,13,28,2,2,100,126,2,2,128,194,1,3,12,30,2,63,270,445,2,0,0,64,6,0,1,9,18,6,7,0,0,5,1,21,7,25,1],[139,74,29,604,746,203,43,0,612,1662,1805,38,0,230,3676,2,0,0,0,0,131,211,178,4,34,0,390,194,1861,323,81,191,125,93,0,28,132,397,598,499,414,1,0,166,1319,117,0,0,11,29,861,65,1008,100,2,0,0,2,29,0,128,80,35,0,0,0,0,274,473,0,0,3,16,91,32,23,13,0,5,19,51,2,7,65,167,1,26,121,132,1,0,8,23,1,65,346,377,0,0,0,75,49,0,2,20,16,23,25,1,5,7,0,29,10,26,0],[132,33,12,360,194,118,8,0,78,281,761,4,0,85,11154,2,0,0,0,0,79,87,39,2,88,0,176,121,781,184,19,74,59,80,0,11,52,164,203,239,176,0,0,91,600,37,0,0,6,13,599,2,265,70,0,0,0,0,5,0,41,85,0,0,0,0,0,91,1136,0,0,6,11,34,12,38,10,0,2,9,9,2,2,48,135,3,15,88,108,1,0,2,6,3,25,82,429,2,0,0,114,23,0,5,6,4,10,2,0,4,0,0,16,6,17,0],[64,18,9,80,50,24,0,0,26,122,102,0,0,18,3301,0,0,0,0,0,34,25,6,0,13,0,32,24,219,51,13,20,26,19,0,1,13,22,37,12,48,0,0,20,106,10,0,0,1,2,121,1,95,21,2,0,0,0,3,0,14,33,0,0,0,0,0,38,1860,8,0,3,4,40,26,13,0,0,1,4,1,5,0,61,259,1,28,147,176,2,3,21,11,0,17,84,662,2,2,1,23,15,0,2,9,3,1,3,1,1,0,0,5,0,8,0],[230,43,70,203,214,62,9,0,78,391,627,0,0,100,9421,5,0,0,0,0,105,43,52,0,56,0,127,79,734,97,18,70,28,61,0,4,33,92,112,178,140,0,0,93,565,17,0,0,1,5,331,4,283,29,0,0,0,0,9,0,33,155,4,0,0,0,0,118,1336,0,0,8,4,27,5,8,8,6,2,10,7,1,4,52,112,1,10,81,106,1,0,8,19,3,37,140,440,2,2,0,66,10,0,7,7,5,13,7,0,3,6,1,16,0,28,2],[120,31,85,171,181,32,1,0,55,273,476,4,0,46,8486,2,0,0,0,0,55,24,24,0,33,0,76,45,291,98,12,42,39,29,0,4,26,57,69,59,62,0,0,51,362,17,0,0,1,5,207,3,195,28,1,0,0,0,4,0,17,67,3,0,0,0,0,86,1236,3,0,1,5,39,14,31,12,0,2,2,18,5,1,48,172,0,14,118,118,0,7,0,8,1,19,116,529,4,0,0,46,18,0,3,3,4,5,6,0,0,2,2,12,2,10,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>M1A<\/th>\n      <th>M1B<\/th>\n      <th>A1A<\/th>\n      <th>A1B<\/th>\n      <th>V1A<\/th>\n      <th>V1B<\/th>\n      <th>M6A<\/th>\n      <th>M6B<\/th>\n      <th>A6A<\/th>\n      <th>A6B<\/th>\n      <th>V6A<\/th>\n      <th>V6B<\/th>\n      <th>M12A<\/th>\n      <th>M12B<\/th>\n      <th>A12A<\/th>\n      <th>A12B<\/th>\n      <th>V12A<\/th>\n      <th>V12B<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"fixedColumns":true,"deferRender":true,"scrollY":200,"scroller":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="data-transformation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-transformation" class="anchor"></a>Data Transformation</h2>
<p>For gene differential expression, raw counts are required, however for data visualization or clustering, it can be useful to work with transformed count data. <code>exploreDDS</code> function is convenience wrapper to transform raw read counts using the <a href="@Love2014-sh"><code>DESeq2</code></a> package transformations methods. The input file has to contain all the genes, not just differentially expressed ones. Supported methods include variance stabilizing transformation (<code>vst</code>) (<span class="citation">Anders and Huber (2010)</span>), and regularized-logarithm transformation or <code>rlog</code> (<span class="citation">Love, Huber, and Anders (2014)</span>).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exploredds</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exploreDDS.html">exploreDDS</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, 
    transformationMethod <span class="op">=</span> <span class="st">"rlog"</span><span class="op">)</span>
<span class="va">exploredds</span></code></pre></div>
<pre><code>## class: DESeqTransform 
## dim: 116 18 
## metadata(1): version
## assays(1): ''
## rownames(116): AT1G01010 AT1G01020 ... ATMG00180 ATMG00200
## rowData names(51): baseMean baseVar ... dispFit rlogIntercept
## colnames(18): M1A M1B ... V12A V12B
## colData names(2): condition sizeFactor</code></pre>
<p>Users are strongly encouraged to consult the <a href="@Love2014-sh"><code>DESeq2</code></a> vignette for more detailed information on this topic and how to properly run <code>DESeq2</code> on data sets with more complex experimental designs.</p>
</div>
<div id="scatterplot" class="section level2">
<h2 class="hasAnchor">
<a href="#scatterplot" class="anchor"></a>Scatterplot</h2>
<p>To decide which transformation to choose, we can visualize the transformation effect comparing two samples or a grid of all samples, as follows:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/exploreDDSplot.html">exploreDDSplot</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, samples <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"M12A"</span>, 
    <span class="st">"M12A"</span>, <span class="st">"A12A"</span>, <span class="st">"A12A"</span><span class="op">)</span>, scattermatrix <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
## `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="systemPipeTools_files/figure-html/exploreDDSplot-1.png" width="700"></p>
<p>The scatterplots are created using the log2 transform normalized reads count, variance stabilizing transformation (VST) (<span class="citation">Anders and Huber (2010)</span>), and regularized-logarithm transformation or <code>rlog</code> (<span class="citation">Love, Huber, and Anders (2014)</span>).</p>
</div>
<div id="hierarchical-clustering-dendrogram" class="section level2">
<h2 class="hasAnchor">
<a href="#hierarchical-clustering-dendrogram" class="anchor"></a>Hierarchical Clustering Dendrogram</h2>
<p>The following computes the sample-wise correlation coefficients using the <code><a href="https://rdrr.io/r/stats/cor.html">stats::cor()</a></code> function from the transformed expression values. After transformation to a distance matrix, hierarchical clustering is performed with the <code><a href="https://rdrr.io/r/stats/hclust.html">stats::hclust</a></code> function and the result is plotted as a dendrogram, as follows:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/hclustplot.html">hclustplot</a></span><span class="op">(</span><span class="va">exploredds</span>, method <span class="op">=</span> <span class="st">"spearman"</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/hclustplot-1.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="hierarchical-clustering-heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#hierarchical-clustering-heatmap" class="anchor"></a>Hierarchical Clustering HeatMap</h2>
<p>This function performs hierarchical clustering on the transformed expression matrix generated within the <code>DESeq2</code> package. It uses, by default, a <code>Pearson</code> correlation-based distance measure and complete linkage for cluster join. If <code>samples</code> selected in the <code>clust</code> argument, it will be applied the <code><a href="https://rdrr.io/r/stats/dist.html">stats::dist()</a></code> function to the transformed count matrix to get sample-to-sample distances. Also, it is possible to generate the <code>pheatmap</code> or <code>plotly</code> plot format.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Samples plot</span>
<span class="fu"><a href="../reference/heatMaplot.html">heatMaplot</a></span><span class="op">(</span><span class="va">exploredds</span>, clust <span class="op">=</span> <span class="st">"samples"</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/heatMaplot_samples-1.png" width="700"></p>
<p>If <code>ind</code> selected in the <code>clust</code> argument, it is necessary to provide the list of differentially expressed genes for the <code>exploredds</code> subset.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Individuals genes identified in DEG analysis DEG analysis with `systemPipeR`</span>
<span class="va">degseqDF</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/run_DESeq2.html">run_DESeq2</a></span><span class="op">(</span>countDF <span class="op">=</span> <span class="va">countMatrix</span>, targets <span class="op">=</span> <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
    independent <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">DEG_list</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/filterDEGs.html">filterDEGs</a></span><span class="op">(</span>degDF <span class="op">=</span> <span class="va">degseqDF</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Fold <span class="op">=</span> <span class="fl">2</span>, FDR <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/heatMaplot_DEG-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### Plot</span>
<span class="fu"><a href="../reference/heatMaplot.html">heatMaplot</a></span><span class="op">(</span><span class="va">exploredds</span>, clust <span class="op">=</span> <span class="st">"ind"</span>, DEGlist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">DEG_list</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/heatMaplot_DEG-2.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="principal-component-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#principal-component-analysis" class="anchor"></a>Principal Component Analysis</h2>
<p>This function plots a Principal Component Analysis (PCA) from transformed expression matrix. This plot shows samples variation based on the expression values and identifies batch effects.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/PCAplot.html">PCAplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/PCAplot-1.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="multidimensional-scaling-with-mdsplot" class="section level2">
<h2 class="hasAnchor">
<a href="#multidimensional-scaling-with-mdsplot" class="anchor"></a>Multidimensional scaling with <code>MDSplot</code>
</h2>
<p>This function computes and plots multidimensional scaling analysis for dimension reduction of count expression matrix. Internally, it is applied the <code><a href="https://rdrr.io/r/stats/dist.html">stats::dist()</a></code> function to the transformed count matrix to get sample-to-sample distances.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/MDSplot.html">MDSplot</a></span><span class="op">(</span><span class="va">exploredds</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/MDSplot-1.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="dimension-reduction-with-glmplot" class="section level2">
<h2 class="hasAnchor">
<a href="#dimension-reduction-with-glmplot" class="anchor"></a>Dimension Reduction with <code>GLMplot</code>
</h2>
<p>This function computes and plots generalized principal components analysis for dimension reduction of count expression matrix.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exploredds_r</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/exploreDDS.html">exploreDDS</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, cmp <span class="op">=</span> <span class="va">cmp</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, preFilter <span class="op">=</span> <span class="cn">NULL</span>, 
    transformationMethod <span class="op">=</span> <span class="st">"raw"</span><span class="op">)</span>
<span class="fu"><a href="../reference/GLMplot.html">GLMplot</a></span><span class="op">(</span><span class="va">exploredds_r</span>, plotly <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/GLMplot-1.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="ma-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#ma-plot" class="anchor"></a>MA plot</h2>
<p>This function plots log2 fold changes (y-axis) versus the mean of normalized counts (on the x-axis). Statistically significant features are colored.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/MAplot.html">MAplot</a></span><span class="op">(</span><span class="va">degseqDF</span>, comparison <span class="op">=</span> <span class="st">"M12-A12"</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Fold <span class="op">=</span> <span class="fl">1</span>, FDR <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, genes <span class="op">=</span> <span class="st">"ATCG00280"</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/MAplot-1.png" width="700"></p>
<p>The function provides the utility to save the plot automatically.</p>
</div>
<div id="t-distributed-stochastic-neighbor-embedding-with-tsneplot" class="section level2">
<h2 class="hasAnchor">
<a href="#t-distributed-stochastic-neighbor-embedding-with-tsneplot" class="anchor"></a>t-Distributed Stochastic Neighbor embedding with <code>tSNEplot</code>
</h2>
<p>This function computes and plots t-Distributed Stochastic Neighbor embedding (t-SNE) analysis for unsupervised nonlinear dimensionality reduction of count expression matrix. Internally, it is applied the <code><a href="https://rdrr.io/pkg/Rtsne/man/Rtsne.html">Rtsne::Rtsne()</a></code> <span class="citation">(Krijthe 2015)</span> function, using the exact t-SNE computing with <code>theta=0.0</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tSNEplot.html">tSNEplot</a></span><span class="op">(</span><span class="va">countMatrix</span>, <span class="va">targets</span>, perplexity <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/tSNEplot-1.png" width="700"></p>
</div>
<div id="volcano-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#volcano-plot" class="anchor"></a>Volcano plot</h2>
<p>A simple function that shows statistical significance (<code>p-value</code>) versus magnitude of change (<code>log2 fold change</code>).</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/volcanoplot.html">volcanoplot</a></span><span class="op">(</span><span class="va">degseqDF</span>, comparison <span class="op">=</span> <span class="st">"M12-A12"</span>, filter <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Fold <span class="op">=</span> <span class="fl">1</span>, FDR <span class="op">=</span> <span class="fl">20</span><span class="op">)</span>, genes <span class="op">=</span> <span class="st">"ATCG00280"</span><span class="op">)</span></code></pre></div>
<p><img src="systemPipeTools_files/figure-html/volcanoplot-1.png" width="700"></p>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version information</h1>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2021-02-04 r79940)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /home/dcassol/src/R-devel/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] systemPipeTools_0.99.3 BiocStyle_2.19.2      
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.2.1             BiocFileCache_1.99.6       
##   [3] systemfonts_1.0.1           plyr_1.8.6                 
##   [5] lazyeval_0.2.2              splines_4.1.0              
##   [7] crosstalk_1.1.1             BiocParallel_1.25.5        
##   [9] GenomeInfoDb_1.27.11        ggplot2_3.3.3              
##  [11] digest_0.6.27               htmltools_0.5.1.1          
##  [13] fansi_0.4.2                 magrittr_2.0.1             
##  [15] checkmate_2.0.0             memoise_2.0.0              
##  [17] BSgenome_1.59.4             base64url_1.4              
##  [19] limma_3.47.12               Biostrings_2.59.2          
##  [21] annotate_1.69.2             matrixStats_0.58.0         
##  [23] systemPipeR_1.25.11         pkgdown_1.6.1              
##  [25] prettyunits_1.1.1           jpeg_0.1-8.1               
##  [27] colorspace_2.0-0            blob_1.2.1                 
##  [29] rappdirs_0.3.3              ggrepel_0.9.1              
##  [31] textshaping_0.3.3           xfun_0.22                  
##  [33] dplyr_1.0.5                 crayon_1.4.1               
##  [35] RCurl_1.98-1.3              jsonlite_1.7.2             
##  [37] genefilter_1.73.1           VariantAnnotation_1.37.1   
##  [39] brew_1.0-6                  survival_3.2-11            
##  [41] ape_5.5                     glue_1.4.2                 
##  [43] gtable_0.3.0                zlibbioc_1.37.0            
##  [45] XVector_0.31.1              DelayedArray_0.17.10       
##  [47] V8_3.4.1                    BiocGenerics_0.37.4        
##  [49] scales_1.1.1                pheatmap_1.0.12            
##  [51] DBI_1.1.1                   GGally_2.1.1               
##  [53] edgeR_3.33.3                Rcpp_1.0.6                 
##  [55] viridisLite_0.4.0           xtable_1.8-4               
##  [57] progress_1.2.2              tidytree_0.3.3             
##  [59] bit_4.0.4                   rsvg_2.1.1                 
##  [61] stats4_4.1.0                DT_0.18                    
##  [63] htmlwidgets_1.5.3           httr_1.4.2                 
##  [65] RColorBrewer_1.1-2          ellipsis_0.3.1             
##  [67] farver_2.1.0                pkgconfig_2.0.3            
##  [69] reshape_0.8.8               XML_3.99-0.6               
##  [71] sass_0.3.1                  dbplyr_2.1.1               
##  [73] locfit_1.5-9.4              utf8_1.2.1                 
##  [75] labeling_0.4.2              later_1.2.0                
##  [77] tidyselect_1.1.0            rlang_0.4.10               
##  [79] AnnotationDbi_1.53.1        munsell_0.5.0              
##  [81] tools_4.1.0                 cachem_1.0.4               
##  [83] generics_0.1.0              RSQLite_2.2.7              
##  [85] evaluate_0.14               stringr_1.4.0              
##  [87] fastmap_1.1.0               yaml_2.2.1                 
##  [89] ragg_1.1.2                  ggtree_2.5.2               
##  [91] knitr_1.33                  bit64_4.0.5                
##  [93] fs_1.5.0                    purrr_0.3.4                
##  [95] KEGGREST_1.31.1             nlme_3.1-152               
##  [97] mime_0.10                   formatR_1.9                
##  [99] aplot_0.0.6                 biomaRt_2.47.8             
## [101] compiler_4.1.0              plotly_4.9.3               
## [103] filelock_1.0.2              curl_4.3                   
## [105] png_0.1-7                   treeio_1.15.7              
## [107] tibble_3.1.1                geneplotter_1.69.0         
## [109] bslib_0.2.4                 stringi_1.5.3              
## [111] highr_0.9                   GenomicFeatures_1.43.8     
## [113] desc_1.3.0                  lattice_0.20-41            
## [115] Matrix_1.3-2                glmpca_0.2.0               
## [117] vctrs_0.3.7                 pillar_1.6.0               
## [119] lifecycle_1.0.0             BiocManager_1.30.12        
## [121] jquerylib_0.1.4             data.table_1.14.0          
## [123] bitops_1.0-7                httpuv_1.6.0               
## [125] rtracklayer_1.51.5          patchwork_1.1.1            
## [127] GenomicRanges_1.43.4        latticeExtra_0.6-29        
## [129] hwriter_1.3.2               R6_2.5.0                   
## [131] BiocIO_1.1.2                promises_1.2.0.1           
## [133] ShortRead_1.49.2            bookdown_0.22              
## [135] IRanges_2.25.10             codetools_0.2-18           
## [137] MASS_7.3-53.1               assertthat_0.2.1           
## [139] SummarizedExperiment_1.21.3 rjson_0.2.20               
## [141] DESeq2_1.31.19              rprojroot_2.0.2            
## [143] withr_2.4.2                 GenomicAlignments_1.27.2   
## [145] Rsamtools_2.7.2             batchtools_0.9.15          
## [147] S4Vectors_0.29.17           GenomeInfoDbData_1.2.4     
## [149] parallel_4.1.0              hms_1.0.0                  
## [151] grid_4.1.0                  tidyr_1.1.3                
## [153] DOT_0.1                     rmarkdown_2.7.12           
## [155] rvcheck_0.1.8               MatrixGenerics_1.3.1       
## [157] Rtsne_0.15                  shiny_1.6.0                
## [159] Biobase_2.51.0              restfulr_0.0.13</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project is funded by NSF award <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1661152">ABI-1661152</a>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Anders2010-tp" class="csl-entry">
Anders, Simon, and Wolfgang Huber. 2010. <span>“Differential Expression Analysis for Sequence Count Data.”</span> <em>Genome Biol.</em> 11 (10): R106.
</div>
<div id="ref-H_Backman2016-bt" class="csl-entry">
H Backman, Tyler W, and Thomas Girke. 2016. <span>“<span class="nocase">systemPipeR: NGS workflow and report generation environment</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.
</div>
<div id="ref-Rtsne" class="csl-entry">
Krijthe, Jesse H. 2015. <em><span>Rtsne</span>: T-Distributed Stochastic Neighbor Embedding Using Barnes-Hut Implementation</em>. <a href="https://github.com/jkrijthe/Rtsne">https://github.com/jkrijthe/Rtsne</a>.
</div>
<div id="ref-Love2014-sh" class="csl-entry">
Love, Michael, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for <span class="nocase">RNA-seq</span> Data with <span>DESeq2</span>.”</span> <em>Genome Biol.</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="https://github.com/mathrj">Ponmathi Ramasamy</a>, <a href="https://lezhang.me/">Le Zhang</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
