<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Visualization Methods for Differential Expression Analysis — exploreDDS • SPtools</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Visualization Methods for Differential Expression Analysis — exploreDDS" />
<meta property="og:description" content="exploreDDS Convenience wrapper function to transform raw read counts using the DESeq2::DESeq2-package() package transformations methods. The input file has to contain all the genes, not just differentially expressed ones.
exploreDDSplot Scatterplot of transformed counts from two samples or grid of all samples.
hclustplot This function computes the sample-wise correlation coefficients using the stats::cor() function from the transformed expression values. After transformation to a distance matrix, hierarchical clustering is performed with the stats::hclust() function, and the result is plotted as a dendrogram.
heatMaplot This function performs hierarchical clustering on the transformed expression matrix. It uses, by default, a Pearson correlation-based distance measure and complete linkage for cluster join.
PCAplot This function plots a Principal Component Analysis (PCA) from transformed expression matrix. This plot shows samples variation based on the expression values and identifies batch effects.
MDSplot This function computes and plots multidimensional scaling analysis for dimension reduction of count expression matrix. Internally, it is applied the stats::dist() function to the transformed count matrix to get sample-to-sample distances.
GLMplot This function computes and plots generalized principal components analysis for dimension reduction of count expression matrix.
MAplot This function plots log2 fold changes (y-axis) versus the mean of normalized counts (on the x-axis). Statistically significant features are colored.
tSNEplot This function computes and plots t-Distributed Stochastic Neighbor embedding (t-SNE) analysis for unsupervised nonlinear dimensionality reduction of count expression matrix. Internally, it is applied the Rtsne::Rtsne() function, using the exact t-SNE computing with theta=0.0.
volcanoplot A simple function that shows statistical significance (p-value) versus magnitude of change (log2 fold change)." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPtools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/systemPipeR_dataViz.html">systemPipeR: Data Visualizations</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Visualization Methods for Differential Expression Analysis</h1>
    
    <div class="hidden name"><code>dataViz.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>exploreDDS</code> Convenience wrapper function to transform raw read counts using the <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeq2-package.html'>DESeq2::DESeq2-package()</a></code> package transformations methods. The input file has to contain all the genes, not just differentially expressed ones.</p>
<p><code>exploreDDSplot</code> Scatterplot of transformed counts from two samples or grid of all samples.</p>
<p><code>hclustplot</code> This function computes the sample-wise correlation coefficients using the <code><a href='https://rdrr.io/r/stats/cor.html'>stats::cor()</a></code> function from the transformed expression values. After transformation to a distance matrix, hierarchical clustering is performed with the <code><a href='https://rdrr.io/r/stats/hclust.html'>stats::hclust()</a></code> function, and the result is plotted as a dendrogram.</p>
<p><code>heatMaplot</code> This function performs hierarchical clustering on the transformed expression matrix. It uses, by default, a Pearson correlation-based distance measure and complete linkage for cluster join.</p>
<p><code>PCAplot</code> This function plots a Principal Component Analysis (PCA) from transformed expression matrix. This plot shows samples variation based on the expression values and identifies batch effects.</p>
<p><code>MDSplot</code> This function computes and plots multidimensional scaling analysis for dimension reduction of count expression matrix. Internally, it is applied the <code><a href='https://rdrr.io/r/stats/dist.html'>stats::dist()</a></code> function to the transformed count matrix to get sample-to-sample distances.</p>
<p><code>GLMplot</code> This function computes and plots generalized principal components analysis for dimension reduction of count expression matrix.</p>
<p><code>MAplot</code> This function plots log2 fold changes (y-axis) versus the mean of normalized counts (on the x-axis). Statistically significant features are colored.</p>
<p><code>tSNEplot</code> This function computes and plots t-Distributed Stochastic Neighbor embedding (t-SNE) analysis for unsupervised nonlinear dimensionality reduction of count expression matrix. Internally, it is applied the <code><a href='https://rdrr.io/pkg/Rtsne/man/Rtsne.html'>Rtsne::Rtsne()</a></code> function, using the exact t-SNE computing with <code>theta=0.0</code>.</p>
<p><code>volcanoplot</code> A simple function that shows statistical significance (<code>p-value</code>) versus magnitude of change (<code>log2 fold change</code>).</p>
    </div>

    <pre class="usage"><span class='fu'>exploreDDS</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, cmp <span class='op'>=</span> <span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, preFilter <span class='op'>=</span> <span class='cn'>NULL</span>, transformationMethod <span class='op'>=</span> <span class='st'>"raw"</span>, blind <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='fu'>exploreDDSplot</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, cmp <span class='op'>=</span> <span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>,preFilter <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>samples</span>, blind <span class='op'>=</span> <span class='cn'>TRUE</span>,  scattermatrix <span class='op'>=</span> <span class='cn'>FALSE</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>
<span class='fu'>hclustplot</span><span class='op'>(</span><span class='va'>exploredds</span>, method <span class='op'>=</span> <span class='st'>"spearman"</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>
<span class='fu'>heatMaplot</span><span class='op'>(</span><span class='va'>exploredds</span>, <span class='va'>clust</span>, DEGlist <span class='op'>=</span> <span class='cn'>NULL</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>
<span class='fu'>PCAplot</span><span class='op'>(</span><span class='va'>exploredds</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>
<span class='fu'>MDSplot</span><span class='op'>(</span><span class='va'>exploredds</span>, method <span class='op'>=</span> <span class='st'>"spearman"</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>
<span class='fu'>GLMplot</span><span class='op'>(</span><span class='va'>exploredds</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>
<span class='fu'>MAplot</span><span class='op'>(</span><span class='va'>exploredds</span>, lfcShrink <span class='op'>=</span> <span class='cn'>FALSE</span>, padj.cutoff <span class='op'>=</span> <span class='fl'>0.05</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span>
<span class='fu'>tSNEplot</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span>, <span class='va'>...</span><span class='op'>)</span>
<span class='fu'>volcanoplot</span><span class='op'>(</span><span class='va'>degseqDF</span>, comparison <span class='op'>=</span> <span class='st'>"M12-A12"</span>, filter <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>Fold <span class='op'>=</span> <span class='fl'>2</span>, FDR <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span>, genes <span class='op'>=</span> <span class='st'>"NULL"</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span>, savePlot <span class='op'>=</span> <span class='cn'>FALSE</span>, filePlot <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>countMatrix</th>
      <td><p><code>date.frame</code> or <code>matrix</code> containing raw read counts.</p></td>
    </tr>
    <tr>
      <th>targets</th>
      <td><p>targets <code>data.frame</code>.</p></td>
    </tr>
    <tr>
      <th>cmp</th>
      <td><p><code>character matrix</code> where comparisons are defined in two columns.
This matrix should be generated with the <code><a href='https://rdrr.io/pkg/systemPipeR/man/readComp.html'>systemPipeR::readComp()</a></code> 
function from the targets file. Values used for comparisons need to match those 
in the <code>Factor</code> column of the targets file.</p></td>
    </tr>
    <tr>
      <th>preFilter</th>
      <td><p>allows removing rows in which there are very few reads. Accepts 
a numeric value with the minimum of total reads to keep. Default is <code>NULL</code>.</p></td>
    </tr>
    <tr>
      <th>transformationMethod</th>
      <td><p>a <code>character string</code> indicating which transformation 
method it will be used on the raw read counts. Supported methods include <code>rlog</code> 
and <code>vst</code> using the <code>DESeq2</code> package or default <code>raw</code> for no data 
transformation.</p></td>
    </tr>
    <tr>
      <th>blind</th>
      <td><p>logical, whether to blind the transformation to the experimental 
design (see varianceStabilizingTransformation), from <code><a href='https://rdrr.io/pkg/DESeq2/man/vst.html'>DESeq2::vst()</a></code> 
or <code><a href='https://rdrr.io/pkg/DESeq2/man/rlog.html'>DESeq2::rlog()</a></code>.</p></td>
    </tr>
    <tr>
      <th>samples</th>
      <td><p>a <code>character vector</code> of two samples or <code>ALL</code> samples in 
the dataset. Could be specified the <code>SampleName</code> column name of the targets 
file or the respective numeric values. Also, if set as <code>ALL</code>, a correlation
matrix it will be plot.</p></td>
    </tr>
    <tr>
      <th>scattermatrix</th>
      <td><p>if <code>samples</code> set as <code>ALL</code>, requires to assign 
<code>TRUE</code> to build a correlation matrix and plot the correlogram of all the samples.</p></td>
    </tr>
    <tr>
      <th>plotly</th>
      <td><p>logical: when <code>FALSE</code> (default), the <code>ggplot2</code> plot will
be returned. <code>TRUE</code> returns the <code>plotly</code> version of the plot.</p></td>
    </tr>
    <tr>
      <th>savePlot</th>
      <td><p>logical: when <code>FALSE</code> (default), the plot will not be saved. 
If <code>TRUE</code> the plot will be saved, and requires the <code>filePlot</code> argument.</p></td>
    </tr>
    <tr>
      <th>filePlot</th>
      <td><p>file name where the plot will be saved. For more information, 
please consult the <code><a href='https://ggplot2.tidyverse.org/reference/ggsave.html'>ggplot2::ggsave()</a></code> function.</p></td>
    </tr>
    <tr>
      <th>exploredds</th>
      <td><p>object of class <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqTransform.html'>DESeq2::DESeqTransform()</a></code>.
<code>GLMplot</code> and <code>MAplot</code>, <code>exploredds</code> object of class <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html'>DESeq2::DESeqDataSet()</a></code>, 
generated from <code>exploreDDS</code> function.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>a <code>character string</code> indicating which correlation coefficient
is to be computed, based on the <code><a href='https://rdrr.io/r/stats/cor.html'>stats::cor()</a></code> function. 
Options are: c("pearson" "kendall", "spearman").</p></td>
    </tr>
    <tr>
      <th>clust</th>
      <td><p>select the data to apply the distance matrix computation. If <code>samples</code> 
selected, it will be applied the <code><a href='https://rdrr.io/r/stats/dist.html'>stats::dist()</a></code> function 
to the transformed count matrix to get sample-to-sample distances. If <code>ind</code>, 
it is necessary to provide the list of differentially expressed genes, for the 
<code>exploredds</code> subset.</p></td>
    </tr>
    <tr>
      <th>DEGlist</th>
      <td><p>List of up or down regulated gene/transcript identifiers meeting
the chosen filter settings for all comparisons defined in data frames <code>pval</code> 
and <code>log2FC</code>.</p></td>
    </tr>
    <tr>
      <th>lfcShrink</th>
      <td><p>logiacal. If <code>TRUE</code> adds shrunken log2 fold changes (LFC) to the object.</p></td>
    </tr>
    <tr>
      <th>padj.cutoff</th>
      <td><p>filter cutoffs for the p-value adjusted.</p></td>
    </tr>
    <tr>
      <th>degseqDF</th>
      <td><p>object of class <code>data.frame</code> generated by <code><a href='https://rdrr.io/pkg/systemPipeR/man/run_edgeR.html'>systemPipeR::run_edgeR()</a></code>
or <code><a href='https://rdrr.io/pkg/systemPipeR/man/run_DESeq2.html'>systemPipeR::run_DESeq2()</a></code>.</p></td>
    </tr>
    <tr>
      <th>comparison</th>
      <td><p><code>character vector</code> specifying the factor names for comparison.</p></td>
    </tr>
    <tr>
      <th>filter</th>
      <td><p>Named vector with filter cutoffs of format c(Fold=2, FDR=1) where
Fold refers to the fold change cutoff (unlogged) and FDR to the p-value cutoff.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p><code>character vecto</code>r of genes names to show on the plot.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>For <code>heatMaplot</code> additional parameters for the <code><a href='https://rdrr.io/pkg/pheatmap/man/pheatmap.html'>pheatmap::pheatmap()</a></code> function.
For <code>GLMplot</code> additional parameters for the <code><a href='https://rdrr.io/pkg/glmpca/man/glmpca.html'>glmpca::glmpca()</a></code> function. 
For <code>tSNEplot</code> additional parameters for the <code><a href='https://rdrr.io/pkg/Rtsne/man/Rtsne.html'>Rtsne::Rtsne()</a></code> function.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>exploreDDS</code> function returns an object of class <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeqTransform.html'>DESeq2::DESeqTransform()</a></code>.
<code>heatMaplot</code> function returns an object of <code>pheatmap</code> or <code>plotly</code> class.
<code>exploreDDSplot</code> function returns an object of <code>ggplot2 plot</code>.
<code>hclustplot</code>, <code>PCAplot</code>, <code>MDSplot</code>, <code>GLMplot</code>, <code>MAplot</code>, 
<code>tSNEplot</code>, and <code>volcanoplot</code> functions returns an object of <code>ggplot</code> or <code>plotly</code> class.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Note that the recommendation is to use the resulting transformed values in the
<code>transformationMethod</code> argument only for visualization and clustering,
not for differential expression analysis which needs raw counts. Users are
strongly encouraged to consult the <code><a href='https://rdrr.io/pkg/DESeq2/man/DESeq2-package.html'>DESeq2::DESeq2-package()</a></code> vignette for more detailed
information on this topic and how to properly run <code>DESeq2</code> on data sets
with more complex experimental designs.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>For more details on <code>DESeq2</code>, please consult the following
page: <a href='http://bioconductor.org/packages/release/bioc/html/DESeq2.html'>DESeq2</a>.
For more details on <code>targets</code> file definition, please consult the following
page: <a href='http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#25_structure_of_targets_file'>systemPipeR</a>.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Daniela Cassol</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>## Targets file</span>
<span class='va'>targetspath</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"targets.txt"</span>, package<span class='op'>=</span><span class='st'>"systemPipeR"</span><span class='op'>)</span>
<span class='va'>targets</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='va'>targetspath</span>, comment<span class='op'>=</span><span class='st'>"#"</span><span class='op'>)</span>
<span class='va'>cmp</span> <span class='op'>&lt;-</span> <span class='fu'>systemPipeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/systemPipeR/man/readComp.html'>readComp</a></span><span class='op'>(</span>file<span class='op'>=</span><span class='va'>targetspath</span>, format<span class='op'>=</span><span class='st'>"matrix"</span>, delim<span class='op'>=</span><span class='st'>"-"</span><span class='op'>)</span>
<span class='co'>## Count table file</span>
<span class='va'>countMatrixPath</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"countDFeByg.xls"</span>, package<span class='op'>=</span><span class='st'>"systemPipeR"</span><span class='op'>)</span>
<span class='va'>countMatrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/utils/read.table.html'>read.delim</a></span><span class='op'>(</span><span class='va'>countMatrixPath</span>, row.names<span class='op'>=</span><span class='fl'>1</span><span class='op'>)</span>

<span class='co'>## Run exploredds</span>
<span class='va'>exploredds</span> <span class='op'>&lt;-</span> <span class='fu'>exploreDDS</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, cmp<span class='op'>=</span><span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, preFilter<span class='op'>=</span><span class='cn'>NULL</span>, transformationMethod<span class='op'>=</span><span class='st'>"rlog"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: some variables in design formula are characters, converting to factors</span></div><div class='input'>
<span class='co'>## Plot exploreDDSplot</span>
<span class='fu'>exploreDDSplot</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, cmp<span class='op'>=</span><span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, preFilter<span class='op'>=</span><span class='cn'>NULL</span>, samples<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span>,<span class='fl'>4</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot hclustplot</span>
<span class='fu'>hclustplot</span><span class='op'>(</span><span class='va'>exploredds</span>, method <span class='op'>=</span> <span class='st'>"spearman"</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot heatMaplot Samples</span>
<span class='fu'>heatMaplot</span><span class='op'>(</span><span class='va'>exploredds</span>, clust<span class='op'>=</span><span class='st'>"samples"</span>, plotly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'>## Individuals genes identified in DEG analysis</span>
<span class='co'>### DEG analysis with `systemPipeR`</span>
<span class='va'>degseqDF</span> <span class='op'>&lt;-</span> <span class='fu'>systemPipeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/systemPipeR/man/run_DESeq2.html'>run_DESeq2</a></span><span class='op'>(</span>countDF <span class='op'>=</span> <span class='va'>countMatrix</span>, targets <span class='op'>=</span> <span class='va'>targets</span>, cmp <span class='op'>=</span> <span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, independent <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: some variables in design formula are characters, converting to factors</span></div><div class='input'><span class='va'>DEG_list</span> <span class='op'>&lt;-</span> <span class='fu'>systemPipeR</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/systemPipeR/man/filterDEGs.html'>filterDEGs</a></span><span class='op'>(</span>degDF <span class='op'>=</span> <span class='va'>degseqDF</span>, filter <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>Fold <span class='op'>=</span> <span class='fl'>2</span>, FDR <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-3.png' alt='' width='700' height='433' /></div><div class='input'><span class='co'>## Plot heatMaplot</span>
<span class='fu'>heatMaplot</span><span class='op'>(</span><span class='va'>exploredds</span>, clust<span class='op'>=</span><span class='st'>"ind"</span>, DEGlist <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/character.html'>as.character</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span><span class='va'>DEG_list</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-4.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot PCAplot</span>
<span class='fu'>PCAplot</span><span class='op'>(</span><span class='va'>exploredds</span>, plotly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'>## Plot MDSplot</span>
<span class='fu'>MDSplot</span><span class='op'>(</span><span class='va'>exploredds</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-5.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot GLMplot</span>
<span class='va'>exploredds_raw</span> <span class='op'>&lt;-</span> <span class='fu'>exploreDDS</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, cmp<span class='op'>=</span><span class='va'>cmp</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>, preFilter<span class='op'>=</span><span class='cn'>NULL</span>, transformationMethod<span class='op'>=</span><span class='st'>"raw"</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='warning'>Warning: some variables in design formula are characters, converting to factors</span></div><div class='input'><span class='fu'>GLMplot</span><span class='op'>(</span><span class='va'>exploredds_raw</span>, plotly <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-6.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot MAplot</span>
<span class='fu'>MAplot</span><span class='op'>(</span><span class='va'>exploredds_raw</span>, plotly <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='output co'>#&gt; [1] "removing NA from the results"</div><div class='input'>
<span class='co'>## Plot tSNEplot</span>
<span class='fu'>tSNEplot</span><span class='op'>(</span><span class='va'>countMatrix</span>, <span class='va'>targets</span>, perplexity <span class='op'>=</span> <span class='fl'>5</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-7.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Plot volcanoplot</span>
<span class='fu'>volcanoplot</span><span class='op'>(</span><span class='va'>degseqDF</span>, comparison <span class='op'>=</span> <span class='st'>"M12-A12"</span>, filter <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>Fold <span class='op'>=</span> <span class='fl'>1</span>, FDR <span class='op'>=</span> <span class='fl'>20</span><span class='op'>)</span>, genes <span class='op'>=</span> <span class='st'>"ATCG00280"</span><span class='op'>)</span>
</div><div class='img'><img src='dataViz-8.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Daniela Cassol, Ponmathi Ramasamy, Le Zhang, Thomas Girke.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


